runtime/domain.c

```diff
@@ -338,12 +338,19 @@ static void caml_wait_interrupt_serviced(struct interruptor* target)
   }
 }
 
-#define MAX_DOMAIN_NAME_LENGTH 16
-void caml_domain_set_name(char *name)
+char caml_pthread_name[MAX_DOMAIN_NAME_LENGTH];
+
+void caml_set_domain_thread_name(char *suffix)
 {
+  char domain_id_suffix[MAX_DOMAIN_NAME_LENGTH - 1] = "";
   char thread_name[MAX_DOMAIN_NAME_LENGTH];
-  snprintf(thread_name, MAX_DOMAIN_NAME_LENGTH,
-           "%s%d", name, Caml_state->id);
+
+  snprintf(domain_id_suffix, MAX_DOMAIN_NAME_LENGTH - 1, ":%s%x",
+           (suffix ? suffix : ""), Caml_state->unique_id);
```

Comment:
Why are you using `MAX_DOMAIN_NAME_LENGTH - 1` instead of `MAX_DOMAIN_NAME_LENGTH` as the limit here? (That is: why is `domain_id_suffix` restricted to be of size `MAX_DOMAIN_NAME_LENGTH - 1`?)

Reply:
So that we have at least 1 character of the original name left, when truncating.